app/gui/controller/double-representation/src/text.rs

```diff
@@ -150,9 +150,11 @@ pub fn apply_code_change_to_id_map(
     }
 
     // If non-preferred entry collides with the preferred one, remove the former.
-    vector.drain_filter(|(range, id)| {
-        preferred.get(range).map(|preferred_id| id != preferred_id).unwrap_or(false)
-    });
+    vector
+        .extract_if(|(range, id)| {
+            preferred.get(range).map(|preferred_id| id != preferred_id).unwrap_or(false)
+        })
+        .for_each(drop);
 }
```

Comment:
This is such a strange change. Did clip told to write it this way? `for_each(drop)` is tacky as hell. The previous version was much more understandable.

Reply:
`extract_if` returns iterator, not using it will cause incorrect behavior. Alternative solution is to use `retain` and flip the condition in the closure, but I was not filling comfortable doing it for such complex expression.